wandb/apis/public/artifacts.py

```diff
@@ -849,6 +851,13 @@ def convert_objects(self):
                 "artifactCollection"
             ]["artifacts"]["edges"]
         ]
+        if self.tags:
+            artifacts = [
+                artifact
+                for artifact in artifacts
+                if not set(self.tags) - set(artifact.tags)
+            ]
+        return artifacts
```

Comment:
suggestion(s): github doesn't let me make a proper inline suggestion here, but consider:

- pulling out `set(self.tags)`, to reduce repeat attribute access
- using `set(a).issubset(b)` or `set(a) <= set(b)` instead of `not set(a) - set(b)`
  - note: I think either suggestion, on balance, might end up being a really tiny perf hit, but (IMO / subjectively) it's reads more clearly
- making the first list expression `artifacts = [...]` into a generator expression `iter_artifacts = (...)`, saving at least an extra traversal of the full list.

(intended to be mainly illustrative, but) in my mind, this would  turn the following snippet: https://github.com/wandb/wandb/blob/509eb41870be5225f720c09827d4e09b297caf33/wandb/apis/public/artifacts.py#L842-L859

...into something roughly like:
```python3
        iter_artifacts = (
            wandb.Artifact._from_attrs(
                self.entity,
                self.project,
                self.collection_name + ":" + a["version"],
                a["node"],
                self.client,
            )
            for a in self.last_response["project"]["artifactType"][
                "artifactCollection"
            ]["artifacts"]["edges"]
        )
        if self.tags:
            expected_tags = set(self.tags)
            artifacts = [
                artifact
                for artifact in iter_artifacts
                if expected_tags.issubset(artifact.tags)
            ]
        else:
            artifacts = list(iter_artifacts)
```

Reply:
Both good suggestions, thanks! I made the second one a generator too so that I don't have to add an `else` clause, but otherwise this is just your suggestion.